src/baal/bayesian/weight_drop.py

```diff
@@ -145,8 +126,11 @@ class MCDropoutConnectModule(torch.nn.Module):
             Name of layers to be replaced from ['Conv', 'Linear', 'LSTM', 'GRU'].
         weight_dropout (float): The probability a weight will be dropped.
     """
+
     def __init__(self, module: torch.nn.Module, layers: Sequence, weight_dropout=0.0):
         super().__init__()
         self.parent_module = module
         _patch_layers(self.parent_module, layers, weight_dropout)
-        self.forward = self.parent_module.forward
+
+    def forward(self, x):
```

Comment:
Does this change make a difference?

Reply:
Yes, by assigning the forward function, DataParallel was not able to clone the model properly and it broke.

Reply:
ah ok!